### DISCUSSION TITLE
select ForeignKey property

### DISCUSSION THREAD

**Discussion by maxiaojunwolf** [aminalaee/sqladmin#329] [General]
The foreign keys must be associated through primary keys and cannot be other unique attributes?

**Comment by aminalaee** [comment]
Can you provide a minimal example and use-case?

**Comment by maxiaojunwolf** [comment]
![image](https://user-images.githubusercontent.com/34365493/191881501-2a89e530-70cb-4071-8c3e-89ae0ca6a0af.png)

The foreign key selection uses the primary key as the ID by default, but my foreign key is not associated through the primary key. When I save data, the ID of the selected data will be updated to the foreign key attribute by default, resulting in a failure to save. For example, if I use a foreign key associated with the unique name attribute, I need to modify the default value of the entry ID to save it as name


class QueryAjaxModelLoader:
    def format(self, model: type) -> Dict[str, Any]:
        if not model:
            return {}

        # return {"id": getattr(model, self.pk.name), "text": str(model)}
        return {"id": getattr(model, self.order_by), "text": str(model)}

**Reply by aminalaee** [reply]
Thanks, I need to take a look. But if you already have a solution, feel free to create a PR.
